### ISSUE TITLE
Produce native image bundles and extract binary

### ISSUE THREAD

**Issue by galderz** [quarkusio/quarkus#36192]
### Description

GraalVM team has asked us that instead of producing a native image binary, we produce a [native image bundle](https://www.graalvm.org/latest/reference-manual/native-image/overview/Bundles/) and then we extract the binary out of there and put it in the expected location. This feature is available since GraalVM 23.0 (GraalVM for JDK 17/20).

**Comment by jerboaa** [comment]
> ### Description
> 
> GraalVM team has asked us that instead of producing a native image binary, we produce a [native image bundle](https://www.graalvm.org/latest/reference-manual/native-image/overview/Bundles/) and then we extract the binary out of there and put it in the expected location. This feature is available since GraalVM 23.0 (GraalVM for JDK 17/20).

I don't think this replaces the native binary. It's more like a replacement/supplement for `quarkus.packaging-type=native-sources` (or whatever the right quarkus flag is for native sources). Bundles can be seen as more like a bill of materials (including environment, flags, configs, etc.)
